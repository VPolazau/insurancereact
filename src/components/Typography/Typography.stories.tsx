import { Typography, TypographyProps } from './Typography';

import type { Meta, Story } from '@storybook/react';

export default {
    tags: ['autodocs'],
    title: 'UI/Typography',
    component: Typography,
} as Meta;

const Template: Story<TypographyProps> = (args) => <Typography {...args} />;

export const Heading1: Story<TypographyProps> = Template.bind({});
Heading1.args = {
    variant: 'h1',
    children: 'Heading 1',
};

export const Heading2: Story<TypographyProps> = Template.bind({});
Heading2.args = {
    variant: 'h2',
    children: 'Heading 1',
};

export const Heading3: Story<TypographyProps> = Template.bind({});
Heading3.args = {
    variant: 'h3',
    children: 'Heading 1',
};

export const Heading4: Story<TypographyProps> = Template.bind({});
Heading4.args = {
    variant: 'h4',
    children: 'Heading 4',
};

export const Heading5: Story<TypographyProps> = Template.bind({});
Heading5.args = {
    variant: 'h5',
    children: 'Heading 5',
};

export const Small: Story<TypographyProps> = Template.bind({});
Small.args = {
    variant: 'small',
    children: 'Small Text',
};

export const Medium: Story<TypographyProps> = Template.bind({});
Medium.args = {
    variant: 'medium',
    children: 'Medium Text',
};

export const Emphasis: Story<TypographyProps> = Template.bind({});
Medium.args = {
    variant: 'emphasis',
    children: 'Emphasis Text',
};
