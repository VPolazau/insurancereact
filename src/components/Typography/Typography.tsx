import React, { HTMLProps } from 'react';
import cn from 'classnames';

import styles from './styles.module.css';

export interface TypographyProps extends HTMLProps<HTMLDivElement> {
    variant: 'h1' | 'h2' | 'h4' | 'h3' | 'h5' | 'medium' | 'small' | 'emphasis' | 'menu' | 'menuBold';
}

export const Typography: React.FC<TypographyProps> = ({ variant, children, className, ...divProps }) => {
    const typographyClasses = cn(styles.text, styles[variant], className);

    return (
        <div className={typographyClasses} data-testid="typographyComponent" {...divProps}>
            {children}
        </div>
    );
};
