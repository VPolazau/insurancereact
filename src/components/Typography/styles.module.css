.h1 {
  font-family: 'Roboto Flex', sans-serif; 
  font-size: 42px;
  font-weight: 600;
  line-height: 130%;
  color: #1F1D1D;
  font-feature-settings: 'liga' off;  
}

.h2 {
  font-family: 'Roboto Flex', sans-serif;
  font-weight: 500;
  font-size: 32px;
  line-height: 130%;
  color: #1F1D1D;
  font-feature-settings: 'liga' off;  
}

.h3 {  
  font-family: 'Roboto Flex', sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 20px;
  line-height: 130%;
  font-feature-settings: 'liga' off;
  color: #1F1D1D;  
}

.h4 {
  font-family: 'Roboto Flex', sans-serif;
  font-size: 17px;
  font-weight: 400;
  line-height: 130%;
  color: #1F1D1D;
  font-feature-settings: 'liga' off;  
}

.h5 {
  font-family: 'Roboto Flex', sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 130%;
  color: #1F1D1D;
  font-feature-settings: 'pnum' on, 'lnum' on, 'liga' off;  
}

.small {
  font-family: 'Nunito Sans', sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 130%;
  color: #1F1D1D;
  font-feature-settings: 'pnum' on, 'lnum' on, 'liga' off;
}

.medium {
  font-family: 'Nunito Sans', sans-serif;
  font-size: 17px;
  font-style: normal;
  font-weight: 400;
  line-height: 130%;
  color: #1F1D1D;
  font-feature-settings: 'liga' off;
}

/* акцент, используется в ЛК */
.emphasis {
  font-family: 'Roboto Flex', sans-serif;
  font-size: 14px;
  font-weight: 600;
  line-height: 130%;
  color: #646464;
  font-feature-settings: 'pnum' on, 'lnum' on, 'liga' off;
}

.menu {
  font-family: 'Roboto Flex', sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 17px;
  line-height: 130%;  
  font-feature-settings: 'liga' off;
  color: #1F1D1D;
}

.menuBold {
  font-family: 'Roboto Flex', sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 17px;
  line-height: 130%;
  font-feature-settings: 'liga' off;
  color: #1F1D1D;
}

@media screen and (max-width: 1150px) {
  .h1 {
    font-size: 38px;
    line-height: 130%;
  }
}

@media screen and (max-width: 965px) {
  .h1 {
    font-size: 32px;
    line-height: 130%;   
    font-weight: 600;
  }
}

@media screen and (max-width: 480px) {
  .h2 {
    font-size: 26px;
  }

  .h3 {
    font-weight: 500;
  }

  .h4 {
    font-size: 15px;
  }
}
