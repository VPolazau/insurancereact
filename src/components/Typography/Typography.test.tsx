import { render, screen } from '@testing-library/react';

import { Typography } from './Typography';

describe('Typography Component', () => {
    it('should render with "h1" variant class', () => {
        const variant = 'h1';
        render(<Typography variant={variant}>Hello, World!</Typography>);
        expect(screen.getByTestId('typographyComponent')).toHaveClass(variant);
    });

    it('should render with "h2" variant class', () => {
        const variant = 'h2';
        render(<Typography variant={variant}>Hello, World!</Typography>);
        expect(screen.getByTestId('typographyComponent')).toHaveClass(variant);
    });

    it('should render with "h3" variant class', () => {
        const variant = 'h3';
        render(<Typography variant={variant}>Hello, World!</Typography>);
        expect(screen.getByTestId('typographyComponent')).toHaveClass(variant);
    });

    it('should render with "h4" variant class', () => {
        const variant = 'h4';
        render(<Typography variant={variant}>Hello, World!</Typography>);
        expect(screen.getByTestId('typographyComponent')).toHaveClass(variant);
    });

    it('should render with "h5" variant class', () => {
        const variant = 'h5';
        render(<Typography variant={variant}>Hello, World!</Typography>);
        expect(screen.getByTestId('typographyComponent')).toHaveClass(variant);
    });

    it('should render with "medium" variant class', () => {
        const variant = 'medium';
        render(<Typography variant={variant}>Hello, World!</Typography>);
        expect(screen.getByTestId('typographyComponent')).toHaveClass(variant);
    });

    it('should render with "small" variant class', () => {
        const variant = 'small';
        render(<Typography variant={variant}>Hello, World!</Typography>);
        expect(screen.getByTestId('typographyComponent')).toHaveClass(variant);
    });

    it('should render with "emphasis" variant class', () => {
        const variant = 'emphasis';
        render(<Typography variant={variant}>Hello, World!</Typography>);
        expect(screen.getByTestId('typographyComponent')).toHaveClass(variant);
    });

    it('should render with correct className', () => {
        const className = 'custom-class';
        render(
            <Typography variant="h4" className={className}>
                Hello, World!
            </Typography>
        );
        expect(screen.getByTestId('typographyComponent')).toHaveClass(className);
    });

    it('should render the correct text', () => {
        const text = 'Hello, World!';
        render(
            <Typography variant="medium" className="custom-class">
                {text}
            </Typography>
        );
        expect(screen.getByText(text)).toBeInTheDocument();
    });

    it('should match the snapshot', () => {
        const variant = 'h1';
        const className = 'custom-class';
        const text = 'Hello, World!';
        const { asFragment } = render(
            <Typography variant={variant} className={className}>
                {text}
            </Typography>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
