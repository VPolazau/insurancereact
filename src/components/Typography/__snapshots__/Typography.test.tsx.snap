// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="text h1 custom-class"
    data-testid="typographyComponent"
  >
    Hello, World!
  </div>
</DocumentFragment>
`;
